integration_tests/src/main/python/join_test.py

```diff
@@ -78,10 +78,13 @@
 # than the number of splits * broadcast threshold and also be at least
 # 3 times smaller than the other side.  So it is not likely to happen
 # unless we can give it some help.
+# To enable replacement of conditional hash joins with nested loop join,
+# set "spark.rapids.sql.replaceConditionalHashJoin.enabled" = "true"
 _hash_join_conf = {'spark.sql.autoBroadcastJoinThreshold': '160',
                    'spark.sql.join.preferSortMergeJoin': 'false',
                    'spark.sql.shuffle.partitions': '2',
-                   'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true'
+                   'spark.sql.legacy.allowNegativeScaleOfDecimal': 'true',
+                   'spark.rapids.sql.replaceConditionalHashJoin.enabled': 'true'
```

Comment:
I think this needs to be a test parameter (`@pytest.mark.parametrize`). We don't want to change the behavior of all tests necessarily, but test with it on and off in specific tests. Thoughts?